src/ImageMetadata.jl

```diff
@@ -164,9 +163,16 @@ Base.delete!(img::ImageMeta, propname::AbstractString) = delete!(img.properties,
 
 # Iteration
 # Defer to the array object in case it has special iteration defined
+if isdefined(Base, :LegacyIterationCompat)
+#    Base.iterate(img::ImageMeta{Array{T,N}}, s=1) where {T,N} = iterate(data(img), s)
+#    Base.iterate(img::ImageMeta, s=(eachindex(img),)) = iterate(data(img), s)
+    Base.iterate(img::ImageMeta) = Base.iterate(data(img))
+    Base.iterate(img::ImageMeta, s) = Base.iterate(data(img), s)
+else
```

Comment:
We could also just get rid of support for `start`/`next`/`done` altogether, but I guess this might be helpful for packages that call these explicitly.